src/controllers/Auth.ts

```diff
@@ -121,8 +134,15 @@ export class AuthController extends BaseController {
    * @param id API key ID
    * @param options Additional options
    *    - `refresh` If set to `wait_for`, Kuzzle will not respond until the API key is indexed
+   *    - `timeout` Request Timeout in ms, after the delay if not resolved the promise will be rejected
    */
-  deleteApiKey(id: string, options: { refresh?: 'wait_for' } = {}): Promise<null> {
+  deleteApiKey(
+    id: string,
+    options: {
+      refresh?: 'wait_for',
+      timeout?: number
```

Comment:
In the future we should take a look at how to handle these 'shared' options for all queries, having 'refresh' and now 'timeout' could just be the beginning of having tons of query oriented options for every API action.

Reply:
You're right, I might be  a good thing to have a better way of handling this kind of shared params, this might need a Workshop